src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4416,21 +4416,30 @@ void StorageReplicatedMergeTree::startupImpl(bool from_attach_thread)
 
         startBeingLeader();
 
-        /// In this thread replica will be activated.
-        restarting_thread.start();
+        if (from_attach_thread)
+        {
+            /// Try activating replica in current thread.
+            restarting_thread.run();
```

Comment:
Don't we need to start it after that?

Reply:
maybe I understood something wrong, but if it was a successful `run`, task will just finish and nothing will be scheduled in background anyway.
In case `run` had a failure, `task->scheduleAfter` is called inside of it so it will be scheduled in both cases.

One thing I would probably need to do is call `activate` on the task so it works with `wakeup`.

Reply:
Yep, the problem is that `schedule`/`scheduleAfter` do nothing if the task was not activated

Reply:
definitely a nice catch